internal/internal_event_handlers.go

```diff
@@ -1105,7 +1106,7 @@ func (weh *workflowExecutionEventHandlerImpl) ProcessEvent(
 	}
 	defer func() {
 		if p := recover(); p != nil {
-			weh.metricsHandler.Counter(metrics.WorkflowTaskExecutionFailureCounter).Inc(1)
```

Comment:
I am a bit surprised this is/was here. @Quinn-With-Two-Ns - are we double-counting this metric for task failures (I see we also increment it at task completion time)?

Reply:
Yes that is called out here https://github.com/temporalio/sdk-go/issues/1450 the fix is to only record this metric in one place. We need to verify in what conditions the current two locations emit a metrics, if there is any duplication or scenarios that are not covered, what other SDKs do, and align Go SDK with other SDKs and only emit the metric in one place.

Reply:
Feel free to toss this PR. Or let me know what you would like me to proceed.

(I am not familiar with other SDKs)

Reply:
Yeah it may be difficult to proceed without knowledge of other SDKs and how the Go SDK processes workflow tasks

Reply:
SG, do you want me to close this PR then?

Reply:
yes please